src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2047,7 +2047,7 @@ private BoundExpression BindSuppressNullableWarningExpression(PostfixUnaryExpres
                 // Report diagnostic if the type consists of no reference types.
                 // PROTOTYPE(NullableReferenceTypes): Rather than checking for reference types,
                 // check whether the suppression is necessary (in NullableWalker) and warn if not.
-                if ((object)type.VisitType((t, a, c) => t.IsErrorType() || t.IsReferenceType || t.IsUnconstrainedTypeParameter(), (object)null, canDigThroughNullable: true) == null)
+                if ((object)type.VisitType((t, a, c) => t.IsErrorType() || !t.IsValueType, (object)null, canDigThroughNullable: true) == null)
```

Comment:
>IsValueType [](start = 78, length = 11)

Do we have test for `t!` where `T t` and `T` is unconstrained?

Reply:
We had a test for `default(T)!` but not for a non-default value. 

---
In reply to: [208670097](https://github.com/dotnet/roslyn/pull/29127#discussion_r208670097) [](ancestors = 208670097)